pkg/cpi/commonUtils.go

```diff
@@ -0,0 +1,65 @@
+package cpi
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/Jeffail/gabs/v2"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/pkg/errors"
+)
+
+//CommonUtils for CPI
+type CommonUtils interface {
+	GetBearerToken() (string, error)
+}
+
+//TokenParameters struct
+type TokenParameters struct {
+	TokenURL, Username, Password string
+	Client                       piperhttp.Sender
+}
+
+// GetBearerToken -Provides the bearer token for making CPI OData calls
+func (tokenParameters TokenParameters) GetBearerToken() (string, error) {
+
+	httpClient := tokenParameters.Client
+
+	clientOptions := piperhttp.ClientOptions{
+		Username: tokenParameters.Username,
+		Password: tokenParameters.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+	tokenFinalURL := fmt.Sprintf("%s?grant_type=client_credentials", tokenParameters.TokenURL)
+	method := "POST"
+	resp, httpErr := httpClient.SendRequest(method, tokenFinalURL, nil, header, nil)
+
```

Comment:
how about checking for `httpErr != nil` here. Then resp should not be empty and the defer can be a one liner:
`defer response.Body.Close()`

This is how it is done throughout the code otherwise, [for example here](https://github.com/SAP/jenkins-library/blob/master/pkg/http/downloader.go#L24-L28).